backend/inspirehep/submissions/views.py

```diff
@@ -160,21 +157,29 @@ def get_updated_record_data(self, update_data, record):
             "arxiv_categories",
             "advisors",
         ]
-        try:
-            updated_ids = update_data["ids"]
-            updated_schemas = get_value(updated_ids, "schema", default=[])
-            updated_orcid = get_values_for_schema(updated_ids, "ORCID")
-            record_ids = record.pop("ids", [])
-            for record_id in record_ids:
-                if (record_id["schema"] not in updated_schemas) or (
-                    record_id["schema"] == "ORCID"
-                    and record_id["value"] != updated_orcid[0]
-                ):
-                    update_data["ids"].append(record_id)
-        except KeyError:
-            pass
 
-        return get_updated_record_data(record, update_data, optional_fields)
+        for field in optional_fields:
+            if field not in set(update_data) and field in set(dict(record)):
+                del record[field]
+
+        for field in update_data:
+            if field == "_private_notes" and record.get(field):
+                record[field].extend(update_data[field])
+            elif field == "ids":
+                record_ids = record.get("ids", [])
+                record_ids_schemas = set(get_value(record, "ids.schema", []))
+                for id_dict in update_data["ids"]:
```

Comment:
we don't have to put it inside `for field in update_data:`. What I would recommend is the following:

Check `_private_notes`
Check `ids`
For loop for fields that are not in `['_private_notes', 'ids']`

By doing that you would avoid the 3 nested loops